src/decanter/core/context.py

```diff
@@ -9,6 +9,14 @@
 
 logger = logging.getLogger(__name__)
 
+def get_or_create_eventloop():
```

Comment:
Looks like this will loop forever?
I tried the change and this will fail when tried to create CoreClient....

Reply:
oh yeah, it does looks like an infinite loop XD